include/nbl/video/alloc/StreamingGPUBufferAllocator.h

```diff
@@ -62,11 +67,15 @@ class StreamingGPUBufferAllocator : protected SimpleGPUBufferAllocator
                 if (allocation.second && (allocation.first->getBoundMemory()->getCurrentMappingCaps()&IDriverMemoryAllocation::EMCAF_READ) )// can read from old
                     memcpy(newAlloc.second,allocation.second+oldOffset_copyRange.first,oldOffset_copyRange.second);
                 else
-                    copyBufferWrapper(allocation.first,newAlloc.first,oldOffset_copyRange.first,0u,oldOffset_copyRange.second);
+                {
+                    // TODO copyBufferWrapper !!!!!!
+                    //copyBufferWrapper(allocation.first,newAlloc.first,oldOffset_copyRange.first,0u,oldOffset_copyRange.second);
+                    assert(false);
+                }
             }
 
             //swap the internals of buffers and book keeping
-            const_cast<IDriverMemoryAllocation*>(allocation.first->getBoundMemory())->unmapMemory();
+            unmapWrapper(const_cast<IDriverMemoryAllocation*>(allocation.first->getBoundMemory()));
             allocation.first->pseudoMoveAssign(newAlloc.first);
             newAlloc.first->drop();
             allocation.second = newAlloc.second;
```

Comment:
does anyone even use `reallocate` ?

Reply:
haven't found anything, almost sure not. I don't get why it even exists tbh

Reply:
ok they probably do, but 
> we should get rid of IGPUBuffer::pseudoMoveAssign and just make the engine work with the fact that SimpleGPUBufferAllocator::reallocate can change the allocation completely

Reply:
hmm okay lets change it to something sane where basically the buffer changes